File: LeetCode/Medium/0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curr_sum = 0, max_sum = INT_MIN;
        for(int val : nums){
            curr_sum += val;
            max_sum = max (max_sum, curr_sum);
            if(curr_sum < 0){
                curr_sum = 0;
            }
        }
        return max_sum;
    }
};

File: LeetCode/Medium/0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
    let r = 0, c = 0;
    for (let i = 0; i < nums.length; i++) {
        c = Math.max(nums[i], c + nums[i]);
        r = Math.max(r, c);
    }
    return r;
};